### DISCUSSION TITLE
Feature: estimated battery life remaining

### DISCUSSION THREAD

**Discussion by austin202220** [sopelj/hass-ember-mug-component#111] [Announcements]
If anyone could help, it would be cool to have an entity that tracks the battery percentage over time to give us an estimated number of minutes remaining before the battery runs out. 

It could be part of the integration, or I guess could be done with a template of some sort in home assistant outside of the integration. Its a bit beyond my skill set to figure out... Thanks!

**Comment by sopelj** [comment]
Hey, I'm definitely open to ideas, but off the top of my head, I can't really think of way to reliably calculate this. The rate at which the battery discharges changes based on many factors and essentially stops when the mug is empty. It doesn't discharge much when cooling, but does quickly when heating. The greater the temperature difference the more energy is used to cool it. Many factors are external too, such as ambient temperature, battery health and such. 

I suppose we could do something very rough if measured how long the battery lasts in each state and then get that time divided by the percent. It would require a lot of testing though and I don't think it would be very reliable.

Personally, I just set up an automation in Home Assistant that warns me if the battery drops below 15% and the mug is not empty. Usually that suffices. 
However, like I said, I'm quite open though.

**Comment by tasomaniac** [comment]
I kinda agree. Even Ember doesn't have this feature. If it was possible, I'm sure they would add this.
